### PR TITLE
Watch changes to history object instead of changed history items

### PR DESCRIPTION
**Pull request by guerler**
So far we have been making continuous request to obtain the most recently changed current history items. This PR switches this to continuous requests on the current history object itself. If the history has changed, the changed items are requested in a separate submission. This keeps the history object in sync and reduces the server workload. It is also a preparation to start using web sockets.

Also transforms the collection and anon user legacy selenium tests to run on the new history.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by mvdbeek — COMMENTED** [review]
The idea is good, but please use an endpoint that returns a 204 if there are no changes since the last request. You can look at the contents_near endpoint for how to do that.

**Comment by guerler** [comment]
@mvdbeek thanks for the review, and yes I can switch the endpoint to use one which returns 204 if no changes have occurred. I wonder if I should do that within this PR or in a follow-up, since this PR also resolves several selenium tests which are critical for the testing coverage as mentioned #13723.

**Review by mvdbeek — APPROVED** [review]

File: lib/galaxy/selenium/navigates_galaxy.py [hunk 1]
```diff
@@ -1389,22 +1389,29 @@ def wait_for_sizzle_selector_clickable(self, selector):
...
-            option_label = option_label_or_component
-            # Click labelled option
-            self.wait_for_visible(self.navigation.history_panel.options_menu)
-            menu_item_sizzle_selector = self.navigation.history_panel.options_menu_item(
-                option_label=option_label
-            ).selector
-            menu_selection_element = self.wait_for_sizzle_selector_clickable(menu_item_sizzle_selector)
-            menu_selection_element.click()
+            if self.is_beta_history():
+                component = self.beta_history_element("share or publish")
```
Comment (jmchilton) [inline lib/galaxy/selenium/navigates_galaxy.py]: this is not right

File: lib/galaxy_test/selenium/test_library_to_collections.py [hunk 2]
```diff
@@ -90,7 +88,8 @@ def list_of_pairs_export(self, is_new_history=False):
         self.collection_builder_create()
         self.home()
         self.history_panel_wait_for_hid_ok(3)
-        self.history_panel_wait_for_hid_hidden(1)
-        self.history_panel_wait_for_hid_hidden(2)
+        # TODO: Check hidden datasets in new history
```
Comment (jmchilton) [inline lib/galaxy_test/selenium/test_library_to_collections.py]: Rebase without this? I think my PR that fixed this has been included in dev now.

File: lib/galaxy/selenium/navigates_galaxy.py (hunk 1 above)
Comment (guerler) [reply lib/galaxy/selenium/navigates_galaxy.py]: Thanks @jmchilton, seeing now that this not used anymore.

File: lib/galaxy_test/selenium/test_library_to_collections.py (hunk 2 above)
Comment (guerler) [reply lib/galaxy_test/selenium/test_library_to_collections.py]: Excellent, thank you so much.

**Review by dannon — APPROVED** [review]

**Comment by jmchilton** [comment]
Thanks for the rebase and the test fixes!

**Pull request merged**